PyInstaller/lib/modulegraph/_compat.py

```diff
@@ -3,7 +3,27 @@
 if sys.version_info[0] == 2:
     def Bchr(value):
         return chr(value)
+    from StringIO import StringIO as BytesIO
+    from StringIO import StringIO
+    from urllib import pathname2url
+    _cOrd = ord
 
+    # File open mode for reading (univeral newlines)
+    _READ_MODE = "rU"
+
+    PY2 = True
 else:
+    from urllib.request import pathname2url
+    from io import BytesIO, StringIO
+    _cOrd = int
+    _READ_MODE = "r"
+
     def Bchr(value):
         return value
+
+    PY2 = False
+
+if sys.version_info < (3, 4):
+    from xdis.std import get_instructions
```

Comment:
So you suggest to use `dis3` for Py2.7, `xdis` for Py3.3 and `dis` for Py3.4+?

I'd simple reduce this to two cases: >=3.4 and <=3.3. For the later I suggest using the already existing code. Only two cases and a few lines of (existing) code.

For the records: [`get_instructions`](https://docs.python.org/3/library/dis.html#dis.get_instructions) is new in Python 3.4.